### PR TITLE
:book: Add vappconfig.md doc

### LINKED ISSUE #123: doc: Update docs/tutorials/deploy-vm/vappconfig.md
This ticket tracks updating [`./docs/tutorials/deploy-vm/vappconfig.md`](https://github.com/vmware-tanzu/vm-operator/tree/main/docs/tutorials/deploy-vm/vappconfig.md). Please refer to https://vm-operator.rtfd.io/en/latest/ref/proj/docs/#project-docs for guidelines on writing VM Operator project documentation.

### PR DESCRIPTION
**Pull request by zyiyi11**
**What does this PR do, and why is it needed?**
Add doc for vAppConfig bootstrap. 

**Which issue(s) is/are addressed by this PR?** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes #123

----
:books: Documentation preview :books:: https://vm-operator--175.org.readthedocs.build/en/175/

### CODE REVIEW

File: docs/tutorials/deploy-vm/vappconfig.md [hunk 1]
```diff
@@ -1,3 +1,89 @@
 # Deploy a VM with vAppConfig
 
-// TODO ([github.com/vmware-tanzu/vm-operator#123](https://github.com/vmware-tanzu/vm-operator/issues/123))
+This page reviews the bootstrap method vAppConfig.
+
+## Description
+The vAppConfig bootstrap method is useful for legacy VM images that rely on bespoke, boot-time processes that leverage vAppConfig properties for customizing a guest.
+
+## Purpose
+As a DevOp, you can assign Golang-based template strings to vAppConfig values to send information into the guest that is not known ahead of time, such as the IPAM data settings used to configure the VM's network stack.
```
Comment (srm09) [inline docs/tutorials/deploy-vm/vappconfig.md:9]:
```suggestion
As a DevOps user, you can assign Golang-based template strings to vAppConfig values to send information into the guest that is not known ahead of time, such as the IPAM data settings used to configure the VM's network stack.
```

File: docs/tutorials/deploy-vm/vappconfig.md [hunk 2]
```diff
@@ -1,3 +1,89 @@
...
+metadata:
+  name: my-secret
+  namespace: test-ns
+stringData:
+  nameservers: "{{ (index .V1alpha1.Net.Nameservers 0) }}"
+  hostname: "{{ .V1alpha1.VM.Name }} "                
+  management_ip: "{{ (index (index .V1alpha1.Net.Devices 0).IPAddresses 0) }}"
+  management_gateway: "{{ (index .V1alpha1.Net.Devices 0).Gateway4 }}"
+```
+1.  :wave: The Golang based template string is a representation of [vm-operator virtualmachinetempl_types v1alpha1 api](https://github.com/vmware-tanzu/vm-operator/blob/25fb865e615d377192a870583ab32973e9fbd32a/api/v1alpha1/virtualmachinetempl_types.go#L4)
```
Comment (srm09) [inline docs/tutorials/deploy-vm/vappconfig.md:38]: Also for the bullets to appear on separate lines, you will need to add a blank line between the two.

File: docs/tutorials/deploy-vm/vappconfig.md [hunk 3]
```diff
@@ -1,3 +1,89 @@
...
+This page reviews the bootstrap method vAppConfig.
+
+## Description
+The vAppConfig bootstrap method is useful for legacy VM images that rely on bespoke, boot-time processes that leverage vAppConfig properties for customizing a guest.
+
+## Purpose
+As a DevOp, you can assign Golang-based template strings to vAppConfig values to send information into the guest that is not known ahead of time, such as the IPAM data settings used to configure the VM's network stack.
+
+## Example
+```
```
Comment (srm09) [inline docs/tutorials/deploy-vm/vappconfig.md:12]:
```suggestion
```yaml
```
☝🏾 For syntax highlighting

File: docs/tutorials/deploy-vm/vappconfig.md [hunk 4]
```diff
@@ -1,3 +1,89 @@
...
+  namespace: test-ns
+stringData:
+  nameservers: "{{ (index .V1alpha1.Net.Nameservers 0) }}"
+  hostname: "{{ .V1alpha1.VM.Name }} "                
+  management_ip: "{{ (index (index .V1alpha1.Net.Devices 0).IPAddresses 0) }}"
+  management_gateway: "{{ (index .V1alpha1.Net.Devices 0).Gateway4 }}"
+```
+1.  :wave: The Golang based template string is a representation of [vm-operator virtualmachinetempl_types v1alpha1 api](https://github.com/vmware-tanzu/vm-operator/blob/25fb865e615d377192a870583ab32973e9fbd32a/api/v1alpha1/virtualmachinetempl_types.go#L4)
+2.  :wave: The fields `nameservers`, `hostname`, `managment_ip` and `management_gateway` are derived from image `haproxy-v0.2.0` OVF properties.
+```
```
Comment (srm09) [inline docs/tutorials/deploy-vm/vappconfig.md:40]:
```suggestion
```xml
```

**Review by srm09 — CHANGES_REQUESTED** [review]
Minor nits, and checkout the preview page generated. Was very helpful to me when reviewing
https://vm-operator--175.org.readthedocs.build/en/175/tutorials/deploy-vm/vappconfig/#supporting-template-queries

File: docs/tutorials/deploy-vm/vappconfig.md [hunk 5]
```diff
@@ -1,3 +1,92 @@
 # Deploy a VM with vAppConfig
 
-// TODO ([github.com/vmware-tanzu/vm-operator#123](https://github.com/vmware-tanzu/vm-operator/issues/123))
+This page reviews the bootstrap method vAppConfig.
+
+## Description
+The vAppConfig bootstrap method is useful for l.For egacy VM images that rely on bespoke, boot-time processes that leverage vAppConfig properties for customizing a guest.
+
+## Purpose
+As a DevOp user, you can assign Golang-based template strings to vAppConfig values to send information into the guest that is not known ahead of time, such as the IPAM data settings used to configure the VM's network stack.
```
Comment (srm09) [inline docs/tutorials/deploy-vm/vappconfig.md:9]:
```suggestion
As a DevOps user, you can assign Golang-based template strings to vAppConfig values to send information into the guest that is not known ahead of time, such as the IPAM data settings used to configure the VM's network stack.
```

**Review by srm09 — CHANGES_REQUESTED** [review]
one final nit, before LGTM

**Review by dilyar85 — APPROVED** [review]
LGTM with some minor suggestions added.

File: docs/tutorials/deploy-vm/vappconfig.md [hunk 6]
```diff
@@ -1,3 +1,92 @@
...
+<Property ovf:key="hostname" ovf:type="string" ovf:userConfigurable="true" ovf:value="ubuntuguest">
+      <Description>Specifies the hostname for the appliance</Description>
+</Property>
+<Property ovf:key="management_ip" ovf:type="string" ovf:userConfigurable="true">
+      <Label>2.3. Management IP</Label>
+      <Description>The static IP address for the appliance on the Management Port Group in CIDR format (.For eg. ip/subnet mask bits). This cannot be DHCP.</Description>
+</Property>
+```
+
+## Supporting Template Queries
```
[the remark this answers is not in the document]
Comment (zyiyi11) [reply docs/tutorials/deploy-vm/vappconfig.md:55]: What do you think if from Guest customization page we link vAppConfig section to this `vappconfig.md`? same goes to cloudinit, sysprep in the future. So in Guest customization page, we don't go into details on how to deploy but focus on the high level summary?

**Comment by srm09** [comment]
@dilyar85 had a comment but otherwise lgtm, approving.

**Review by srm09 — APPROVED** [review]

File: docs/tutorials/deploy-vm/vappconfig.md [hunk 7]
```diff
@@ -1,3 +1,92 @@
...
+    - if input netmask is different with CIDR, replace and return IP with new CIDR. 
+    - if input netmask is empty string, return IP sans CIDR. 
+    - if input netmask is not valid, return empty string. 
+
+**Note** when OVF only takes in IP sans CIDR, use `V1alpha1_FormatIP` and pass empty string as input netmask:
+```yaml
+management_ip: '{{ V1alpha1_FormatIP V1alpha1_FirstIP "" }}' # return first non-loopback IP from first NIC without CIDR. For eg, "192.168.1.10".
+```
+
+### Examples:
```
[the remark this answers is not in the document]
Comment (zyiyi11) [reply docs/tutorials/deploy-vm/vappconfig.md:105]: Sounds good. I added a vAppConfig with templating queries and also kept this part.

File: docs/tutorials/deploy-vm/vappconfig.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (dilyar85) [reply docs/tutorials/deploy-vm/vappconfig.md:78]: Yep, that sounds good to me too.

**Pull request merged**